mesonbuild/interpreterbase.py

```diff
@@ -642,19 +662,20 @@ def evaluate_division(self, l, r):
             return l // r
         raise InvalidCode('Division works only with strings or integers.')
 
-    def evaluate_arithmeticstatement(self, cur):
+    def evaluate_arithmeticstatement(self, cur: mparser.ArithmeticNode) -> Union[int, str, dict, list, Disabler]:
```

Comment:
I think this is fine for now, but I think that the various `Node` classes really are generics and should be annotated as such.

Reply:
Not all of them are (at least according to my limited knowledge regarding generics in python). For instance, the `CodeBlockNode` always stores a list of `BaseNode`.

Reply:
I have annotated the `ElementaryNode` as a generic. I don't see how the other nodes could be generics.

Reply:
I think some of them are specializations:

```python

class Node(typing.Generic[T])

    held: T

    def get(self) -> T:
         return held


class StrNode(Node[str]):
    pass
```

Reply:
Yes, they should already be annotated in `mparser.py`.